pkg/nsx/services/subnet/subnet.go

```diff
@@ -459,3 +461,148 @@ func (service *SubnetService) UpdateSubnetSet(ns string, vpcSubnets []*model.Vpc
 	}
 	return nil
 }
+
+// GetNSXSubnetByAssociatedResource gets the NSX subnet based on the associated resource annotation
+func (service *SubnetService) GetNSXSubnetByAssociatedResource(ns string, associatedResource string, vpcService common.VPCServiceProvider) (*model.VpcSubnet, error) {
+	// Parse the associated resource string (format: projectID:vpcID:subnetID)
+	parts := strings.Split(associatedResource, ":")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("invalid associated resource format: %s, expected format: projectID:vpcID:subnetID", associatedResource)
+	}
+
+	projectID := parts[0]
+	vpcID := parts[1]
+	subnetID := parts[2]
+
+	// Get VPC info for the namespace
+	vpcInfoList := vpcService.ListVPCInfo(ns)
+	if len(vpcInfoList) == 0 {
+		return nil, fmt.Errorf("no VPC info found for namespace: %s", ns)
+	}
+
+	vpcInfo := vpcInfoList[0]
```

Comment:
We have a discussion here https://github.com/vmware-tanzu/nsx-operator/pull/1096#discussion_r2115095589 that we can use orgID as default so that no need to introduce vpcService here.